src/formats/jpeg.zig

```diff
@@ -0,0 +1,1084 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+const image = @import("../image.zig");
+const FormatInterface = @import("../format_interface.zig").FormatInterface;
+const color = @import("../color.zig");
+const PixelFormat = @import("../pixel_format.zig").PixelFormat;
+const ImageReader = image.ImageReader;
+const ImageSeekStream = image.ImageSeekStream;
+const ImageFormat = image.ImageFormat;
+const ImageInfo = image.ImageInfo;
+
+// TODO: Chroma subsampling
+// TODO: Progressive scans
+// TODO: Non-baseline sequential DCT
+// TODO: Precisions other than 8-bit
+
+const JPEG_DEBUG = false;
+const JPEG_VERY_DEBUG = false;
+
+// Marker codes
+
+const Markers = enum(u16) {
+    // Start of Frame markers, non-differential, Huffman coding
+    SOF0 = 0xFFC0, // Baseline DCT
+    SOF1 = 0xFFC1, // Extended sequential DCT
+    SOF2 = 0xFFC2, // Progressive DCT
+    SOF3 = 0xFFC3, // Lossless sequential
+    // Start of Frame markers, differential, Huffman coding
+    SOF5 = 0xFFC5, // Differential sequential DCT
+    SOF6 = 0xFFC6, // Differential progressive DCT
+    SOF7 = 0xFFC7, // Differential lossless sequential
+    // Start of Frame markers, non-differential, arithmetic coding
+    SOF9 = 0xFFC9, // Extended sequential DCT
+    SOF10 = 0xFFCA, // Progressive DCT
+    SOF11 = 0xFFCB, // Lossless sequential
+    // Start of Frame markers, differential, arithmetic coding
+    SOF13 = 0xFFCD, // Differential sequential DCT
+    SOF14 = 0xFFCE, // Differential progressive DCT
+    SOF15 = 0xFFCF, // Differential lossless sequential
+
+    DefineHuffmanTables = 0xFFC4,
+    DefineArithmeticCoding = 0xFFCC,
+
+    StartOfImage = 0xFFD8,
+    EndOfImage = 0xFFD9,
+    StartOfScan = 0xFFDA,
+    DefineQuantizationTables = 0xFFDB,
+    DefineNumberOfLines = 0xFFDC,
+    DefineRestartInterval = 0xFFDD,
+    DefineHierarchicalProgression = 0xFFDE,
+    ExpandReferenceComponents = 0xFFDF,
+
+    // Add 0-15 as needed.
+    Application0 = 0xFFE0,
+    // Add 0-13 as needed.
+    JpegExtension0 = 0xFFF0,
+    Comment = 0xFFFE,
+};
+
+const DensityUnit = enum {
+    Pixels,
+    DotsPerInch,
+    DotsPerCm,
+};
+
+const JFIFHeader = struct {
+    jfifRevision: u16,
```

Comment:
Use Zig coding convention here, `jfif_revision`.

Reply:
Done, this was my first piece of serious Zig code where I hadn't figured out the coding conversion yet :P